Quarkus (#737)
vaadin/docs: articles/flow/integrations/quarkus/basic.asciidoc

---
title: Quarkus
order: 1
layout: page
---

[[quarkus.basic]]
= Using Quarkus with Vaadin

Quarkus is an open source, Kubernetes-native Java framework that allows 
developers to optimize container-native applications for peak performance.
Check https://quarkus.io/ to get more info about Quarkus.

> To be able to run Quarkus  the `vaadin-quarkus` and `vaadin-core-jandex` dependencies 

Author: I have no idea

I've just checked the base starter structure and put here the vaadin plugins which I found there.
If you say that it's not needed let's remove it.
Should I just remove it or replace with something ?

> To be able to run Quarkus  the `vaadin-quarkus` and `vaadin-core-jandex` dependencies 

Reviewer: well at the moment it is needed as I don't think it's part of the vaadin(-core) dependencies, but perhaps we should just say vaadin-jandex as it works with both (vaadin/vaadin-core) cases and doesn't have an impact on the size of the compiled project.

> To be able to run Quarkus  the `vaadin-quarkus` and `vaadin-core-jandex` dependencies 

Author: done

are required.
The Quarkus application can be run using `quarkus-maven-plugin` using `quarkus:dev` goal.

A base starter project https://github.com/vaadin/base-starter-flow-quarkus/ can be used
as a starting point for your own application.

== Vaadin CDI Features

Quarkus is a CDI container so it's possible to use all CDI features.

> Check the CDI documentation for these features:

Reviewer: ```suggestion
See the documentation for Vaadin CDI features:
```

- <<../cdi/contexts#, Vaadin CDI Scopes>> 

> - <<../cdi/events#, Observable Vaadin Eventss>> 

Reviewer: ```suggestion
- <<../cdi/events#, Observable Vaadin Events>> 
```

- <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>

> - <<../cdi/nstantiated-beans#, Using CDI Beans>>

Reviewer: ```suggestion
- <<../cdi/instantiated-beans#, Using CDI Beans>>
```

[[quarkus.vaadin.addons]]

> = Vaadin add-ons in Quarkus Application

Reviewer: ```suggestion
== Vaadin add-ons In Quarkus Applications
```
Deployment seems to not accept main headers inside the text...

> The Vaadin add-on used in Quarkus should have jandex index. This can be done via `jandex-maven-plugin`.

Reviewer: ```suggestion
Any Vaadin add-on used in a Quarkus application should contain a jandex index.
This is generated using the `jandex-maven-plugin`.
See https://quarkus.io/guides/cdi-reference#how-to-generate-a-jandex-index[How to Generate a Jandex Index]
```

> If you can’t modify the dependency, you can still index it by adding `quarkus.index-dependency` 
> entries to your application.properties:

Reviewer: ```suggestion
If you can’t modify the dependency, you can still get Quarkus to index it by adding `quarkus.index-dependency` 
entries to your application.properties:
```

.application.properties
[source, properties]
----
quarkus.index-dependency.<name>.group-id=
quarkus.index-dependency.<name>.artifact-id=
quarkus.index-dependency.<name>.classifier=(this one is optional)
----

[[quarkus.vaadin.limitations]]
== Limitations

> Websockets Push doesn't work currently in Vaadin integration for Quarkus.

Reviewer: ```suggestion
WebSockets Push doesn't work currently in the Vaadin extension for Quarkus.
```

---
title: Quarkus
order: 1
layout: page
---

[[quarkus.basic]]
= Using Quarkus With Vaadin

Quarkus is an open source, Kubernetes-native Java framework made for Java virtual machines and native compilation, optimizing Java specifically for containers and enabling it to become an effective platform for serverless, cloud, and Kubernetes environments.
Check https://quarkus.io/ to get more info about Quarkus.

To be able to run Quarkus the `vaadin-quarkus` and `vaadin-jandex` dependencies 
are required.
Using the `quarkus-maven-plugin` the `quarkus:dev` goal is used to start the application.

> The https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] project  can be used
> as a starting point for your own application.

Reviewer: ```suggestion
The https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] project can be used
as a starting point for your own application.
There is also the https://github.com/vaadin/base-starter-flow-quarkus/tree/gradle[base starter for gradle].
```

== Vaadin CDI Features

Quarkus’ dependency injection solution is based on CDI, so it's possible to use all CDI features.

> See the documentation for Vaadin CDI features:
> * <<../cdi/contexts#, Vaadin CDI Scopes>> 

Reviewer: ```suggestion
See the documentation for Vaadin CDI features:

* <<../cdi/contexts#, Vaadin CDI Scopes>> 
```
Need new line else seen as one line?

> - <<../cdi/contexts#, Vaadin CDI Scopes>> 
> - <<../cdi/events#, Observable Vaadin Events>> 
> - <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>
> - <<../cdi/instantiated-beans#, Using CDI Beans>>

Reviewer: ```suggestion
* <<../cdi/contexts#, Vaadin CDI Scopes>> 
* <<../cdi/events#, Observable Vaadin Events>> 
* <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>
* <<../cdi/instantiated-beans#, Using CDI Beans>>
```
I guess these should be with `*`s as they are not getting a new line in the preview: https://deploy-preview-737--vaadin-docs-preview.netlify.app/flow/integrations/quarkus/basic/#vaadin-cdi-features
